code/modules/antagonists/hivemind/spells/hivemind_spells.dm

```diff
@@ -0,0 +1,42 @@
+/obj/effect/proc_holder/spell/target_hive
+	panel = "Hivemind Abilities"
+	invocation_type = "none"
+	selection_type = "range"
+	action_icon = 'icons/mob/actions/actions_hive.dmi'
+	action_background_icon_state = "bg_hive"
+	action_icon_state = "spell_default"
+	clothes_req = 0
+	human_req = 1
+	antimagic_allowed = TRUE
+	range = 0 //SNOWFLAKE, 0 is unlimited for target_external=0 spells
+	var/target_external = 0 //Whether or not we select targets inside or outside of the hive
+
+
+/obj/effect/proc_holder/spell/target_hive/choose_targets(mob/user = usr)
+	var/datum/antagonist/hivemind/hive = user.mind.has_antag_datum(/datum/antagonist/hivemind)
+	if(!hive || !hive.hivemembers)
+		to_chat(user, "<span class='notice'>This is a bug. Error:HIVE1</span>")
+		return
+	var/list/possible_targets = list()
+	var/list/targets = list()
+
+	if(target_external)
+		for(var/mob/living/carbon/H in view_or_range(range, user, selection_type))
+			if(user == H)
+				continue
+			if(!can_target(H))
+				continue
+			if(!hive.is_carbon_member(H))
+				possible_targets += H
+	else
+		possible_targets = hive.get_carbon_members()
+		if(range)
+			possible_targets &= view_or_range(range, user, selection_type)
+
+	var/mob/living/carbon/human/H = input("Choose the target for the spell.", "Targeting") as null|mob in possible_targets
+	if(!H)
+		revert_cast()
+		return
+	targets += H
+	perform(targets,user=user)
```

Comment:
I think these should be a regular target spell but fail if the person isn't in the hive.

Reply:
Except for the ones like hive vision actually

Reply:
I believe the ones that should be targeted I've already swapped over